src/core/tsi/s2a/handshaker/s2a_tsi_handshaker.h

```diff
@@ -21,64 +21,78 @@
+ *  - is_client: a boolean that is true if |client| is used at the client side,
+ *    and false if |client| is used at the server side.
+ *  - interested_parties: set of pollsets interested in this connection.
+ *  - self: the address of S2A TSI handshaker instance to be populated by the
+ *    method; the caller must ensure that |self| is not nullptr.
+ *  - error_details: an error message for when the creation fails. It is legal
+ *    (and expected) to have |error_details| point to a nullptr.
  *
  * It returns TSI_OK on success and an error status code on failure. Note that
  * if interested_parties is nullptr, a dedicated TSI thread will be created and
  * used. **/
 tsi_result s2a_tsi_handshaker_create(
     const grpc_s2a_credentials_options* options, const char* target_name,
-    const char* handshaker_service_url, bool is_client,
+    bool is_client,
     grpc_pollset_set* interested_parties, tsi_handshaker** self,
     char** error_details);
 
 /** This method creates an S2A TSI handshaker result instance.
  *  - response: the data received from the S2A handshaker service.
  *  - is_client: a boolean that is true if |client| is used at the client side,
  *    and false if |client| is used at the server side.
- *  - result: the address of the S2A TSI handshaker result instance to be
+ *  - self: the address of the S2A TSI handshaker result instance to be
  *    populated by the method. **/
 tsi_result s2a_tsi_handshaker_result_create(s2a_SessionResp* response,
                                             bool is_client,
-                                            tsi_handshaker_result** result);
+                                            tsi_handshaker_result** self);
 
 /** This method sets the unused bytes of an S2A TSI handshaker result instance.
- * - result: an S2A TSI handshaker result instance.
- * - recv_bytes: data received from the handshaker service.
- * - bytes_consumed: size of data consumed by the handshaker service. **/
-void s2a_tsi_handshaker_result_set_unused_bytes(tsi_handshaker_result* result,
+ *  - self: an S2A TSI handshaker result instance; the caller must not pass in
+ *    nullptr for this argument.
+ *  - recv_bytes: data received from the handshaker service; the caller must not
+ *    pass in nullptr for this argument.
+ *  - bytes_consumed: size of data consumed by the handshaker service. **/
+void s2a_tsi_handshaker_result_set_unused_bytes(tsi_handshaker_result* self,
                                                 grpc_slice* recv_bytes,
                                                 size_t bytes_consumed);
 
 /** This method returns a boolean value indicating whether or not an
  *  s2a_tsi_handshaker instance has been shutdown. **/
 bool s2a_tsi_handshaker_has_shutdown(s2a_tsi_handshaker* handshaker);
 
+/** The following two methods are exposed for testing purposes only. **/
+void s2a_check_tsi_handshaker(tsi_handshaker* base, grpc_slice target_name,
```

Comment:
Could you please add `_for_testing` suffix for these two methods?

Reply:
Done.